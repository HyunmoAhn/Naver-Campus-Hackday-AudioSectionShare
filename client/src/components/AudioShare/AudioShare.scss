.AudioShare {

}

.AudioShare__content {
  color: deepskyblue;
  margin: auto;
  width: 80%;
  border: 1px solid deepskyblue;
  border-radius: 15px;
  margin-bottom: 15px;
}

.AudioShare__textarea {
  display: block;
  margin: auto;
  width: 80%;
}

.AudioShare__btn {
  margin: 15px;
  height: 30px;
}

.AudioShare__icon {
  width: 100%;
  height: 100%;
}