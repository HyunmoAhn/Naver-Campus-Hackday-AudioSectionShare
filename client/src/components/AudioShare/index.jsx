import React from 'react';
import PropTypes from 'prop-types';
import './AudioShare.scss';

const propTypes = {
	shareContent: PropTypes.string,
	onFacebookShare: PropTypes.func,
	onNaverShare: PropTypes.func,
};

const defaultProps = {
	shareContent: '',
	onFacebookShare() {},
	onNaverShare() {},
};

class AudioShare extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			value: '',
		};
	}

	render() {
		if (this.props.shareContent) {
			return (
				<div className="AudioShare__content">
					{this.props.shareContent}
				</div>
			)
		}
		return (
			<div className="AudioShare">
				<textarea
					className="AudioShare__textarea"
					placeholder="공유 시 전달하고 싶은 문구를 적어주세요."
					cols="30"
					rows="10"
				/>
				<button
					className="AudioShare__btn AudioShare__btn-naver"
				  type="button"
				  onClick={this.props.onNaverShare}
				>
					Naver
				</button>
				<button
					className="AudioShare__btn AudioShare__btn-facebook"
				  type="button"
				  onClick={this.props.onFacebookShare}
				>
					Facebook
				</button>
			</div>
		);
	}
}

AudioShare.propTypes = propTypes;
AudioShare.defaultProps = defaultProps;

export default AudioShare;