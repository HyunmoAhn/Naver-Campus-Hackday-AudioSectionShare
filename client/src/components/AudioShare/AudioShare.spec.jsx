import React from 'react';
import { shallow } from 'enzyme';

import AudioShare from './';

describe('<AudioShare />', () => {
	const mockId = 'mock-id';
	const mockStartTime = 15;
	const mockEndTime = 30;

	it('should match to snapthot when render default', () => {
		const wrapper = shallow(<AudioShare
			id={mockId}
			startTime={mockStartTime}
			endTime={mockEndTime}
		/>);

		expect(wrapper).toMatchSnapshot();
	});

	it('should setState value when onChange textarea', () => {
		const mockValue = 'mock-value';
		const wrapper = shallow(<AudioShare
			id={mockId}
		  startTime={mockStartTime}
		  endTime={mockEndTime}
		/>);
		const textarea = wrapper.find('textarea');

		textarea.simulate('change', {
			target: {
				value: mockValue,
			},
		});
		expect(wrapper.state().value).toEqual(mockValue);
	});

	it('should call onNaverShare when click .AudioShare__btn-naver', () => {
		const onNaverShare = jest.fn();
		const wrapper = shallow(<AudioShare
			id={mockId}
		  startTime={mockStartTime}
		  endTime={mockEndTime}
		  onNaverShare={onNaverShare}
		/>);
		const naverBtn = wrapper.find('.AudioShare__btn-naver');
		const { id, startTime, endTime } = wrapper.instance().props;
		const { value } = wrapper.state();

		naverBtn.simulate('click');
		expect(onNaverShare.mock.calls.length).toBe(1);
		expect(onNaverShare.mock.calls[0]).toEqual([id, startTime, endTime, value]);
	});

	it('should call onFacebookShare when click .AudioShare__btn-facebook', () => {
		const onFacebookShare = jest.fn();
		const wrapper = shallow(<AudioShare
			id={mockId}
		  startTime={mockStartTime}
		  endTime={mockEndTime}
		  onFacebookShare={onFacebookShare}
		/>);
		const facebookBtn = wrapper.find('.AudioShare__btn-facebook');
		const { id, startTime, endTime } = wrapper.instance().props;
		const { value } = wrapper.state();

		facebookBtn.simulate('click');
		expect(onFacebookShare.mock.calls.length).toBe(1);
		expect(onFacebookShare.mock.calls[0]).toEqual([id, startTime, endTime, value]);
	});
});