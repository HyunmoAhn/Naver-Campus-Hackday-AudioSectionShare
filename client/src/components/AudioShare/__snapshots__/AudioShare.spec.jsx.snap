// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioShare /> should match to snapthot when render default 1`] = `
<div
  className="AudioShare"
>
  <textarea
    className="AudioShare__textarea"
    cols="30"
    onChange={[Function]}
    placeholder="공유 시 전달하고 싶은 문구를 적어주세요."
    rows="10"
    value=""
  />
  <button
    className="AudioShare__btn AudioShare__btn-naver"
    onClick={[Function]}
    type="button"
  >
    Naver
  </button>
  <button
    className="AudioShare__btn AudioShare__btn-facebook"
    onClick={[Function]}
    type="button"
  >
    Facebook
  </button>
</div>
`;
